### PR TITLE
decode template util func added for used in other test

### PR DESCRIPTION
**Pull request by nurali-techie**

### CODE REVIEW

File: pkg/test/template/util_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	err := apis.AddToScheme(s)
+	require.NoError(t, err)
+	codecFactory := serializer.NewCodecFactory(s)
+	decoder := codecFactory.UniversalDeserializer()
+
+	t.Run("valid_template", func(t *testing.T) {
+		// test
+		tmpl, err := DecodeTemplate(decoder, validTmplContent)
+
+		assert.NoError(t, err)
```
Comment (xcoulon) [inline pkg/test/template/util_test.go]: you should use 
```suggestion
		require.NoError(t, err)
``` 
here instead

File: pkg/test/template/util_test.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		assert.NotNil(t, tmpl)
```
Comment (xcoulon) [inline pkg/test/template/util_test.go]: same here: use 
```suggestion
		require.NotNil(t, tmpl)
```

File: pkg/test/template/util.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+		return nil, fmt.Errorf("decoder cannot be nil")
+	}
+	obj, _, err := decoder.Decode(tmplContent, nil, nil)
+	if err != nil {
+		return nil, errs.Wrapf(err, "unable to decode template")
+	}
+	tmpl, ok := obj.(*templatev1.Template)
+	if !ok {
+		return nil, fmt.Errorf("unable to convert object type %T to Template, must be a v1.Template", obj)
+	}
```
Comment (xcoulon) [inline pkg/test/template/util.go]: actually, you can do it in a simpler way, without having to convert the returned object into a `*templatev1.Template`:

```suggestion
tmplObj := &templatev1.Template{}
_, _, err = g.decoder.Decode(tmpl, nil, tmplObj)
if err != nil {
	return nil, errs.Wrapf(err, "unable to decode template")
}
```

File: pkg/test/template/util_test.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+		assert.Equal(t, "mytemplate", tmpl.GetName())
+		assert.NotNil(t, tmpl.Objects)
```
Comment (xcoulon) [inline pkg/test/template/util_test.go]: same here:
```suggestion
		require.NotNil(t, tmpl.Objects)
```

File: pkg/test/template/util_test.go [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+		assert.Equal(t, 1, len(tmpl.Objects))
+	})
+
+	t.Run("invalid_template", func(t *testing.T) {
+		// test
+		tmpl, err := DecodeTemplate(decoder, invalidTmplContent)
+
+		assert.Error(t, err)
```
Comment (xcoulon) [inline pkg/test/template/util_test.go]:
```suggestion
		require.Error(t, err)
```

File: pkg/test/template/util_test.go [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+		assert.Nil(t, tmpl)
+		assert.Contains(t, err.Error(), "unable to decode template")
+	})
+
+	t.Run("nil_decoder", func(t *testing.T) {
+		// test
+		tmpl, err := DecodeTemplate(nil, invalidTmplContent)
+
+		assert.Error(t, err)
```
Comment (xcoulon) [inline pkg/test/template/util_test.go]:
```suggestion
		require.Error(t, err)
```

File: pkg/test/template/util_test.go [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+		assert.Nil(t, tmpl)
+		assert.Contains(t, err.Error(), "decoder cannot be nil")
+	})
+
+	t.Run("nil_template", func(t *testing.T) {
+		// test
+		tmpl, err := DecodeTemplate(decoder, nil)
+
+		assert.Error(t, err)
```
Comment (xcoulon) [inline pkg/test/template/util_test.go]:
```suggestion
		require.Error(t, err)
```

File: pkg/test/template/util.go (hunk 3 above)
Comment (nurali-techie) [reply pkg/test/template/util.go]: @xcoulon with this simpler way I don't thing this wrapper function required in `common/test`.  We can directly call `decoder.Decode(tmplContent, nil, tmplObj)` in our test_code wherever required .. WDYS ??

Comment (xcoulon) [reply pkg/test/template/util.go]: right, the function body is pretty straightforward, indeed, and this func here adds no real benefit, so we can probably close this PR.

Comment (nurali-techie) [reply pkg/test/template/util.go]: Thanks for simpler way suggestion :-)  Closing this PR.

**Pull request closed without merging**